nvme-ioctl.c

```diff
@@ -393,39 +394,55 @@ int nvme_get_log(int fd, __u32 nsid, __u8 log_id, __u32 data_len, void *data)
 	__u16 numdu = numd >> 16, numdl = numd & 0xffff;
 
 	cmd.cdw10 = log_id | (numdl << 16);
+	if (lsp)
+                cmd.cdw10 |= ((int)lsp) << 8;
```

Comment:
Q. why _lsp_ has to be casted to int first?

Reply:
I meant to take that out. I wasn't sure of the integer promotion rules here. For example, does the shift promote up to 32bit then cast back down or is shifting by the width of a u8 on a u8 undefined behavior? 
I didn't know the answer to this, and I didn't want to open the c-spec or go read assembly so I cast it to int. If you know the answer, I'll change it.